### ISSUE TITLE
Add script to login to erp during runtime while extracting the data.

### ISSUE THREAD

**Issue by Ayushk4** [kshitij10496/hercules#1]
As of now, we have to pass the cookie as an argument while running the extract data scripts. 

We can use a web scraper /web driver ( selenium or robobrowser ) and login into erp while running the script. This will prompt the user to input the roll number and the password and the security question. Note that this won't be storing any password or other details that were input This will save us the trouble of having to log in and then copy paste the cookies.

This will make it easier for running the scripts to update every sem.

**Comment by kshitij10496** [comment]
Sounds like a good idea. 😃 

I think we _should_ have capabilities to perform ERP login in `hercules` in the future. 
Since the code is open-sourced, the user shouldn’t be afraid/paranoid to share their credentials.
We had a similar terminal based credentials sharing system in `gyft`.
The advantages provided by this login service would be wholesome in itself with people easily able to their personalised data on ERP through a more user-friendly API. 

However, this is just my personal opinion and I'm very much interested in listening to what others' have to say before we start implementing anything here. 😅 

cc KGP Community, to name a few: @amrav @hargup @icyflame @ghostwriternr

**Comment by icyflame** [comment]
@kshitij10496 Good reasoning there. 👍  As long as the script that the user is running is short and can be inspected by them, asking them to enter their password is completely fine! 🙂

**Comment by kshitij10496** [comment]
> @kshitij10496 Making an environment variable for the method will let the user choose his/her way of feeding either the credentials or the cookies IMHO.

That's a good idea!
At the moment, the scripts expect a `JSESSIONID` as a positional argument for scrapping.
To be honest, it would be wonderful if we can have the best of both the worlds! 🤔 

```sh
$ python scrapper.py JSESSIONID -o results.json # scrapping with Session ID

$ python scrapper.py -o results.json # scrapping with login
Enter your Roll No.
Enter your Password
[Security Question]
Enter your Answer:
```

**Comment by kshitij10496** [comment]
Closed by #2
